core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -353,6 +359,17 @@ public Builder removeProperties(Set<String> propertiesToRemove) {
       return this;
     }
 
+    public ViewMetadata.Builder assignUUID(String newUUID) {
+      Preconditions.checkArgument(newUUID != null, "Cannot set uuid to null");
+
+      if (null == uuid) {
```

Comment:
In Iceberg, we don't want to silently ignore user input. In this case, if the caller tries to reassign to a different UUID then this should fail with a good error message about how you can't reassign a view UUID.

Reply:
makes sense, I've added a precondition check that fails when a user tries to reassign the UUID